mongodbatlas/fw_provider.go

```diff
@@ -0,0 +1,394 @@
+				Description: "Amazon Resource Names (ARNs) of IAM Policies describing further restricting permissions for the IAM Role being assumed.",
+			},
+			"role_arn": schema.StringAttribute{
+				Optional:    true,
+				Description: "Amazon Resource Name (ARN) of an IAM Role to assume prior to making API calls.",
+			},
+			"session_name": schema.StringAttribute{
+				Optional:    true,
+				Description: "An identifier for the assumed role session.",
+				Validators: []validator.String{
+					stringvalidator.LengthBetween(2, 64),
+					stringvalidator.RegexMatches(regexp.MustCompile(`[\w+=,.@\-]*`), ""),
+				},
+			},
+			"source_identity": schema.StringAttribute{
+				Optional:    true,
+				Description: "Source identity specified by the principal assuming the role.",
+				Validators: []validator.String{
+					stringvalidator.LengthBetween(2, 64),
+					stringvalidator.RegexMatches(regexp.MustCompile(`[\w+=,.@\-]*`), ""),
+				},
+			},
+			"tags": schema.MapAttribute{
+				ElementType: types.StringType,
+				Optional:    true,
+				Description: "Assume role session tags.",
+			},
+			"transitive_tag_keys": schema.SetAttribute{
+				ElementType: types.StringType,
+				Optional:    true,
+				Description: "Assume role session tag keys to pass to any subsequent sessions.",
+			},
+		},
+	},
+}
+
+func (p *MongodbtlasProvider) Configure(ctx context.Context, req provider.ConfigureRequest, resp *provider.ConfigureResponse) {
+	var (
+		data tfMongodbAtlasProviderModel
+	)
+
+	resp.Diagnostics.Append(req.Config.Get(ctx, &data)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	defaultValuesAndValidations(&data, resp)
+
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	config := Config{
+		PublicKey:    data.PublicKey.ValueString(),
+		PrivateKey:   data.PrivateKey.ValueString(),
+		BaseURL:      data.BaseURL.ValueString(),
+		RealmBaseURL: data.RealmBaseURL.ValueString(),
+	}
+
+	var assumeRoles []tfAssumeRoleModel
+	data.AssumeRole.ElementsAs(ctx, &assumeRoles, true)
+
+	if len(assumeRoles) > 0 {
+		config.AssumeRole = parseTfModel(ctx, &assumeRoles[0])
+		secret := data.SecretName.ValueString()
+		region := data.Region.ValueString()
+		awsAccessKeyID := data.AwsAccessKeyID.ValueString()
+		awsSecretAccessKey := data.AwsSecretAccessKeyID.ValueString()
+		awsSessionToken := data.AwsSessionToken.ValueString()
+		endpoint := data.StsEndpoint.ValueString()
+		var err error
+		config, err = configureCredentialsSTS(&config, secret, region, awsAccessKeyID, awsSecretAccessKey, awsSessionToken, endpoint)
+		if err != nil {
+			resp.Diagnostics.AddError("failed to configure credentials STS", err.Error())
+			return
+		}
+	}
+
+	client, err := config.NewClient(ctx)
+
+	if err != nil {
+		resp.Diagnostics.AddError(
+			"failed to initialize a new client",
+			err.Error(),
+		)
+		return
+	}
+
+	resp.DataSourceData = client
+	resp.ResourceData = client
+}
+
+func parseTfModel(ctx context.Context, tfAssumeRoleModel *tfAssumeRoleModel) *AssumeRole {
```

Comment:
Could you add a comment to explain what is the goal of the function?

Reply:
This is not addressed

Reply:
adjusted